torchx/workspace/docker_workspace.py

```diff
@@ -132,6 +137,16 @@ def build_workspace_and_update_role(
             role.image = image.id
         finally:
             context.close()
+        if len(old_imgs) > 0 and role.image in old_imgs:
```

Comment:
instead of moving the logging message from `torchx.runner.api` (which affects users using other workspace implementations). Could we do this check 3 lines above and not change the `role.image` to `image.id` if we know that the images are the same?

Reply:
Good idea, did as requested. If the image sha hash is detected locally we leave role.image unchanged ( to trigger the Reused message), if the image sha is not detected we assign the sha to role.image(triggering the new build message).

Reply:
awesome, thanks for the change!